cborg/src/Codec/CBOR/FlatTerm.hs

```diff
@@ -77,8 +78,10 @@ data TermToken
     | TkInteger                 !Integer
     | TkBytes    {-# UNPACK #-} !ByteString
     | TkBytesBegin
-    | TkString   {-# UNPACK #-} !Text
+    | TkByteArray     {-# UNPACK #-} !BA.ByteArray
+    | TkString        {-# UNPACK #-} !Text
     | TkStringBegin
+    | TkUtf8ByteArray {-# UNPACK #-} !BA.ByteArray
```

Comment:
I suppose we don't strictly speaking need these. Eliminating them would require converting from a `ByteArray` to a `Text`/`ByteString` in `toFlatTerm`, which is a bit annoying but perhaps acceptable.